docs/html-templates.md

```diff
@@ -0,0 +1,40 @@
+# Html templates
+
+### Behavior
+
+* Templates are automatically recompiled in development when the file
+is saved; a refresh of the page should be enough to display the
+latest version.
+
+* The context, which sets the values that the variables in HTML
+templates should take, is defined in `zerver/context_processors.py`.
+Additional variable-setting might take place in the render call for
+that specific template.
+
+## Backend
+
+For text generated in the backend, including logged-out ("portico")
+pages and the webapp's base content, we use the [Jinja2][] template
+engine.
```

Comment:
added info for backend & frontend, not sure if i covered all the locations though